backend/core/src/core/video/application/video-service.ts

```diff
@@ -123,6 +127,57 @@ export class VideoService {
     return this.videoRepository.addVideoCommentReply(request, userId);
   }
 
+  //NOTE: remove segments and poster from cloud
+  async deleteByIds(ids: string[]): Promise<VideoInfoDto[] | null> {
+    const deletedVideos = await this.videoRepository.deleteByIds(ids);
+    if (!deletedVideos.length) {
+      return null;
+    }
+
+    return deletedVideos.map((video) => castToVideoInfoDto(video));
+  }
+
+  async getMyVideos(authorId: string): Promise<VideoInfoDto[]> {
+    const videos = await this.videoRepository.getMyVideos(authorId);
+
+    return videos.map((video) => castToVideoInfoDto(video));
+  }
+
+  async updateVisibility({ videoId, visibility }: UpdateVideoVisibilityDto): Promise<VideoInfoDto | null> {
+    const isVideoExists = await this.videoRepository.getById(videoId);
+    if (!isVideoExists) {
+      return null;
+    }
+    const video = await this.videoRepository.updateVisibility({
+      videoId,
+      visibility,
+    });
+
+    if (!video) {
+      return null;
+    }
```

Comment:
I guess you can remove first call to repository (`getById`), if this one covers your needs (for checking if video existed)